website/content/project/chiventure.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "chiventure: a Text Adventure Game Engine"
+type: page
+---
+
+```
+You wake up to find yourself in a strange hall. You have no idea how you got here. The last thing you remember is a blinding flash of light. You look at the tapestries lining the walls for any clues on your whereabouts. The tapestries depict valiant knights, fire-breathing wyverns, armies of goblins, and fearsome rulers sitting in their gilded thrones. Wherever you are, it must be in a different world from your own. 
+
+No, not just a different world, but a different time altogether. 
+
+A time of wizards and warlords and all manner of magical creatures, a time to fight back the forces of Throxar the Terrible and to heed the counsel of Wilfrax the Wise... 
+
+It is a time of adventure. 
+
+It is the time of...
+```
+
+[![chiventure](/cmsc22000/img/chiventure.jpg "chiventure")](https://www.youtube.com/watch?v=DGSXURPvRxY)
+
+
+The CMSC 22000 project revolves around implementing a _text adventure game engine_. This means that you won’t be implementing an individual game but, rather, a platform that allows game authors (with little or no knowledge of computer programming) to develop text adventure games.
+
+Not just that, you won't be implementing it from scratch but, rather, will work on existing code that was written in previous editions of CMSC 22000.
+
+
+# Text Adventure Games
+
+An [adventure game](https://en.wikipedia.org/wiki/Adventure_game) is a type of video game where the player is the protagonist of an interactive story. Broadly speaking, the protagonist inhabits a world that they can interact with: exploring their surroundings (e.g., reading a sign on the wall), going from one part of the world to another (e.g., going from one room to an adjacent room), manipulating objects in the world (e.g., using a crowbar to open a door), or conversing with non-player characters to gain knowledge that may prove useful later on. The game will have some sort of objective (like finding a treasure), which will mark the conclusion of the game (and the story).
+
+In the early days of video games, adventure games were _text-based_, and players would specify actions using a command-line interface. For example, here is a screenshot of the classic adventure game [Zork](https://en.wikipedia.org/wiki/Zork) (note how commands are entered after the `>` prompt)
+
+
+![Zork](/cmsc22000/img/zork.png "Zork")
+
+
+Over time, adventure games became more sophisticated, featuring graphical interfaces, like the one in [The Secret of Monkey Island](https://en.wikipedia.org/wiki/The_Secret_of_Monkey_Island) (1990):
+
+
+![The Secret of Monkey Island](/cmsc22000/img/monkey_island.png "The Secret of Monkey Island")
```

Comment:
Maybe add a link to the game's [Steam page](https://store.steampowered.com/wishlist/profiles/76561198033270289) so that we can directly point students who want to play a text-based game? Getting access to the other games mentioned may be difficult for them (not sure how easy it is to set up emulators, etc.), but I think this one is a good example for them to check out.

Reply:
This reminds me that, last year, Ben Collins-Sussman suggested that we point students to  the interactive spy novel, [Spider and Web](https://eblong.com/zarf/zweb/tangle/). I'm going to add a link to that, along to Ben's [Introduction to Interactive Fiction](https://www.red-bean.com/sussman/if/) page.

Adding a link to Monkey Island on Steam sounds good too.